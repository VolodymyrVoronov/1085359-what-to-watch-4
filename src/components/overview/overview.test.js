import React from "react";
import renderer from "react-test-renderer";
import Overview from "./overview.jsx";

const CURRENT_MOVIE = {
  id: 0,
  title: `Citizen Kane`,
  genre: `Drama`,
  img: `img/bg-the-grand-budapest-hotel.jpg`,
  releaseDate: 2014,

  poster: `img/bg-the-grand-budapest-hotel.jpg`,
  rating: 7.8,
  ratingReviewsCount: 10,
  description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  director: `Director`,
  actors: [
    `Actor 1`,
    `Actor 2`,
    `Actor 3`,
    `Actor 4`,
    `Actor 5`,
  ],
  runTime: 200,
  preview: ``,

  reviews: [
    {
      author: `John`,
      score: 5.1,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    },
    {
      author: `Max`,
      score: 4.1,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    },
    {
      author: `Stive`,
      score: 10,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    },
    {
      author: `Stive`,
      score: 10,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    }
  ]
};

it(`render should be match markup`, () => {

  const result = renderer
    .create(<Overview
      film={CURRENT_MOVIE}
    />)
    .toJSON();

  expect(result).toMatchSnapshot();
});
