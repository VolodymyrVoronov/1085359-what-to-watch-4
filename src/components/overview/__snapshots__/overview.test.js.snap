// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should be match markup 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        10 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Actor 1, Actor 2, Actor 3, Actor 4, Actor 5
      </strong>
    </p>
  </div>,
]
`;
